docs/guides/installing-tls-certificates.md

```diff
@@ -0,0 +1,97 @@
+# Generating and installing TLS certificates
+
+- [1. Overview](#1-overview)
+- [2. Generating TLS certificates](#2-generating-tls-certifificates)
+- [3. Installing TLS certificates](#3-installing-tls-certificates)
+    - [3.1 Install at default location](#31-install-at-default-location)
+    - [3.2 Install at non-default location](#32-install-at--non--default-location)
+
+## 1. Overview
+
+This document provides information about generating and installing TLS certificates
+for running infrap4d in secure mode
+
+## 2. Generating TLS certificates
+Note: Here target name can be `dpdk` or `es2k`
+
+- Review the files `ca.conf` and `grpc-client.conf` available under
+`/usr/share/stratum/<target_name>` to verify that the configuration settings are
+as desired
+- Run the `generate-certs.sh` available under `/usr/share/stratum/<target_name>`
+
+```bash
+	cd /usr/share/stratum/<target_name>/
+
+	COMMON_NAME=<IP> ./generate-certs.sh
+	or
+	COMMON_NAME=localhost ./generate-certs.sh
+```
+
+Note: System relies on mTLS (mutual TLS) for authentication.
+
+## 3. Installing TLS certificates
+
+`infrap4d` will check for server certificates in the default location
+`/usr/share/stratum/certs/`
+
+### 3.1 Install at default location
+
+- Copy the generated `ca.crt`, `stratum.crt`, and `stratum.key` to the server
+running infrap4d:
+
+
+```bash
+mkdir -p /usr/share/stratum/certs
```

Comment:
I don't understand the context.

Why copy the certificates from `/usr/share/stratum/<target_name>/certs` to `/usr/share/stratum/certs` on the same platform? Why not simply generate them there? Better still, why not have the script and configuration files in `/usr/share/stratum/certs` generate the key and certificate files in place?

If you're copying them to a _different_ platform, there's nothing in the instructions that make that celar.

Reply:
This is how the code and scripts were written initially.

generate-certs.sh will generate certs folder having all certs. This will be copied to /usr/share/stratum/certs which is the default location.

I have said "Copy the generated `ca.crt`, `stratum.crt`, and `stratum.key` to the server running infrap4d". In our case, it can he host or ACC. Same applies for client too.